### PR TITLE
[CARBONDATA-40]	Make the derby db metastore configurable

### PR DESCRIPTION
**Pull request by ravipesala**
Details are given in https://issues.apache.org/jira/browse/CARBONDATA-40

### CODE REVIEW

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala [hunk 1]
```diff
@@ -32,8 +34,22 @@ import org.carbondata.common.logging.LogServiceFactory
...
-class CarbonContext(val sc: SparkContext, val storePath: String) extends HiveContext(sc) {
+class CarbonContext(
+    val sc: SparkContext,
+    val storePath: String,
+    metaStorePath: String) extends HiveContext(sc) with Logging {
   self =>
+
+  def this (sc: SparkContext) = {
+    this (sc,
+      new File("./carbonstore").getCanonicalPath,
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala]: there is a configuration CarbonCommonConstants.STORE_LOCATION, that should be taken as default value when storelocation is not provided.

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala [hunk 2]
```diff
@@ -32,8 +34,22 @@ import org.carbondata.common.logging.LogServiceFactory
...
+      new File("./carbonmetastore").getCanonicalPath)
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala]: we also can add a configuration  CarbonCommonConstants.META_STORE_LOCATION, defaulting to ./carbonmetastore.

**Comment by gvramana** [comment]
Other than configuration comments, remaining LGTM.

**Comment by ravipesala** [comment]
Fixed review comments

**Comment by QiangCai** [comment]
Verified in local spark shell.

**Comment by QiangCai** [comment]
Good. I also find this compile error in test case. I will cancel my modification.

**Comment by QiangCai** [comment]
I think this pr is important to simplify QuickStart.
@ravipesala can you refresh [QuickStart](https://github.com/apache/incubator-carbondata/blob/master/docs/Quick-Start.md)?

**Comment by ravipesala** [comment]
@chenliang613 Please review and merge this PR.

**Comment by chenliang613** [comment]
verified , LGTM

**Pull request closed without merging**